packages/renderer/src/lib/network/NetworksList.spec.ts

```diff
@@ -0,0 +1,281 @@
+  );
+
+  render(NetworksList);
+  await tick();
+
+  const network1Name = screen.getByRole('cell', { name: 'Network 1' });
+  const network2Name = screen.getByRole('cell', { name: 'Network 2' });
+  expect(network1Name).toBeInTheDocument();
+  expect(network2Name).toBeInTheDocument();
+
+  expect(network1Name.compareDocumentPosition(network2Name)).toBe(4);
+});
+
+test('Expect to have edit action for Podman networks', async () => {
+  vi.mocked(window.getProviderInfos).mockResolvedValue([providerInfoMock]);
+
+  vi.mocked(window.listNetworks).mockResolvedValue([network2, network1]);
+
+  window.dispatchEvent(new CustomEvent('extensions-already-started'));
+  window.dispatchEvent(new CustomEvent('provider-lifecycle-change'));
+
+  // wait for stores to be populated
+  await waitFor(
+    () => {
+      expect(get(providerInfos)).not.toHaveLength(0);
+      expect(get(networksListInfo)).not.toHaveLength(0);
+    },
+    { timeout: 2000 },
+  );
+
+  render(NetworksList);
+  await tick();
+
+  const editButtons = screen.getAllByRole('button', { name: 'Update Network' });
+  expect(editButtons.length).toBe(1);
+
+  const podmanNetworkRow = screen.getByRole('row', { name: 'Network 2' });
+  expect(podmanNetworkRow).toBeInTheDocument();
+  expect(podmanNetworkRow).toContain(editButtons[0]);
+});
+
+test('Expect to have delete action for unused networks', async () => {
+  vi.mocked(window.getProviderInfos).mockResolvedValue([providerInfoMock]);
+
+  vi.mocked(window.listNetworks).mockResolvedValue([network2, network1]);
+
+  window.dispatchEvent(new CustomEvent('extensions-already-started'));
+  window.dispatchEvent(new CustomEvent('provider-lifecycle-change'));
+
+  // wait for stores to be populated
+  await waitFor(
+    () => {
+      expect(get(providerInfos)).not.toHaveLength(0);
+      expect(get(networksListInfo)).not.toHaveLength(0);
+    },
+    { timeout: 2000 },
+  );
+
+  render(NetworksList);
+  await tick();
+
+  const deleteButtons = screen.getAllByRole('button', { name: 'Delete Network' });
+  expect(deleteButtons.length).toBe(1);
+
+  const unusedNetworkRow = screen.getByRole('row', { name: 'Network 1' });
+  expect(unusedNetworkRow).toBeInTheDocument();
+  expect(unusedNetworkRow).toContain(deleteButtons[0]);
+});
+
+test('Expect user confirmation for bulk delete when required', async () => {
+  vi.mocked(window.getProviderInfos).mockResolvedValue([providerInfoMock]);
+
+  vi.mocked(window.listNetworks).mockResolvedValue([network2, network1]);
+
+  window.dispatchEvent(new CustomEvent('extensions-already-started'));
+  window.dispatchEvent(new CustomEvent('provider-lifecycle-change'));
+
+  // wait for stores to be populated
+  await waitFor(
+    () => {
+      expect(get(providerInfos)).not.toHaveLength(0);
+      expect(get(networksListInfo)).not.toHaveLength(0);
+    },
+    { timeout: 2000 },
+  );
+
+  render(NetworksList);
+  await tick();
+
+  const checkboxes = screen.getAllByRole('checkbox', { name: 'Toggle network' });
```

Comment:
```suggestion
  const checkboxes = screen.getAllByRole('checkbox', { name: 'Toggle network' });
  expect(checkboxes).toHaveLength(1)
```

suggestion(non-blocking): probably nice to ensure the array contain the item